game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/MockGameData.java

```diff
@@ -155,4 +161,34 @@ public MockGameData withCaptureUnitsOnEnteringTerritory(final boolean value) {
     when(gameProperties.get(CAPTURE_UNITS_ON_ENTERING_TERRITORY, false)).thenReturn(value);
     return this;
   }
+
+  public MockGameData withEditMode(final boolean value) {
```

Comment:
Side-note, I don't know if this pattern will scale well. Ideally we'd just use a real object here. To do that, extracting the game properties to it's own class and then passing that properties class around would perhaps be the thing to do. To that extent, we could have the battle logic take in that properties object and not a full game data as well for further benefits.

The reason I don't think this will scale is because eventually with more test coverage we'll have re-implemented the full game properties object and have a near duplicate dummy implementation.